react/PlaceOrder.tsx

```diff
@@ -1,14 +1,138 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { FormattedMessage } from 'react-intl'
 import { Button } from 'vtex.styleguide'
+import { OrderForm } from 'vtex.order-manager'
+import postRobot from 'post-robot'
+import { useRuntime } from 'vtex.render-runtime'
+import { OrderPayment } from 'vtex.order-payment'
 
-import styles from './PlaceOrder.css'
+const { useOrderForm } = OrderForm
+const { useOrderPayment } = OrderPayment
 
 const PlaceOrder: React.FC = () => {
+  const { orderForm } = useOrderForm()
+  const { value, referenceValue, interestValue } = useOrderPayment()
+  const {
+    culture: { currency },
+    rootPath = '',
+  } = useRuntime()
+
+  const [placingOrder, setPlacingOrder] = useState(false)
+  const [cardFormIframe] = useState(
+    () => document.getElementById('chk-card-form')! as HTMLIFrameElement
+  )
+
+  const handlePlaceOrder = async () => {
+    const transactionData = {
+      referenceId: orderForm.id,
+      value,
+      referenceValue,
+      interestValue,
+      savePersonalData: true,
+      optinNewsLetter: false,
+    }
+
+    setPlacingOrder(true)
+
+    const startTransactionResponse = await fetch(
```

Comment:
Will you move this to a Graphql call in the future?

Reply:
no, I don't think it would benefit much here :-/